ui/src/components/DashboardMetrics.js

```diff
@@ -0,0 +1,460 @@
+          title: `${element.name}-read`,
+          formatType: 'throughputFormatter',
+        });
+        res.push({
+          field: `${element.name}-write`,
+          type: 'quantitative',
+          title: `${element.name}-write`,
+          format: '.2f',
+        });
+      });
+      return res;
+    })(nodes),
+  );
+
+  const lineConfig = { strokeWidth: 1.5 };
+
+  const formatNodesPromRangeForChart = useCallback(
+    (result) => {
+      const reduced = nodes.reduce((acc, node, index) => {
+        let temp = [];
+        if (
+          result[index].status === 'success' &&
+          result[index].data.result[0].values.length
+        ) {
+          temp = result[index].data.result[0].values.map((item) => ({
+            date: fromUnixTimestampToDate(item[0]),
+            type: node.name,
+            y: item[1],
+          }));
+        }
+        return acc.concat(temp);
+      }, []);
+      return reduced;
+    },
+    [nodes],
+  );
+
+  const formatNodesThroughputPromRangeForChart = useCallback(
+    (result) => {
+      const readRes = result[0];
+      const writeRes = result[1];
+
+      if (readRes.status === 'success' && writeRes.status === 'success') {
+        const reduced = nodes.reduce((acc, node, index) => {
+          let tempRead = [];
+          let tempWrite = [];
+
+          const nodeReadData = readRes.data?.result?.find(
+            (item) => item.metric?.instance.split(':')[0] === node.internalIP,
+          );
+          const nodeWriteData = writeRes.data?.result?.find(
+            (item) => item.metric?.instance.split(':')[0] === node.internalIP,
+          );
+
+          if (nodeReadData)
+            tempRead = nodeReadData.values.map((item) => ({
+              date: fromUnixTimestampToDate(item[0]),
+              type: `${node.name}-read`,
+              y: 0 - item[1],
+            }));
+          if (nodeWriteData)
+            tempWrite = nodeWriteData.values.map((item) => ({
+              date: fromUnixTimestampToDate(item[0]),
+              type: `${node.name}-write`,
+              y: item[1],
+            }));
+
+          return acc.concat(tempRead).concat(tempWrite);
+        }, []);
+        return reduced;
+      }
+
+      return null;
+    },
+    [nodes],
+  );
+
+  // App config, used to generated Advanced metrics button link
+  const configQuery = useQuery('appConfig', fetchConfig);
+
+  // Passing nodes table as a react-queries identifier so if a node is added/removed the data are refreshed
+  // Also it makes the data to auto-refresh based on the node refresh timeout that is already implemented
+  const cpuDataQuery = useQuery(
+    ['dashboardMetricsCPU', nodes, metricsTimeSpan],
+    useCallback(
+      () =>
+        Promise.all(
+          nodes.map((node) =>
+            queryNodeCPUMetrics(node.internalIP, metricsTimeSpan),
+          ),
+        ).then((result) => formatNodesPromRangeForChart(result)),
+      [nodes, metricsTimeSpan, formatNodesPromRangeForChart],
```

Comment:
what's the reason we put `formatNodesPromRangeForChart` as useCallback dependency?

Reply:
@ChengYanJin to avoid the warning: 
` React Hook useCallback has a missing dependency: 'formatNodesPromRangeForChart'. Either include it or remove the dependency array  react-hooks/exhaustive-deps`

React wants to know it needs to update the function if formatNodesPromRangeForChart changes (it won't happen) but it considers it's a good practice to include it.